3rdparty/openssl/deprecations.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_OPENSSL_DEPRECATIONS_H
+#define _OE_OPENSSL_DEPRECATIONS_H
+
+#if !defined(OE_OPENSSL_SUPPRESS_DEPRECATIONS)
+
+#if defined(__cplusplus)
+#define OE_OPENSSL_EXTERN_C_BEGIN \
+    extern "C"                    \
+    {
+#define OE_OPENSSL_EXTERN_C_END }
+#else
+#define OE_OPENSSL_EXTERN_C_BEGIN
+#define OE_OPENSSL_EXTERN_C_END
+#endif
+
+#define OE_OPENSSL_DEPRECATED(MSG) __attribute__((deprecated(MSG)))
+
+OE_OPENSSL_EXTERN_C_BEGIN
+
+/*
+**==============================================================================
+**
+** OE_UNSUPPORTED_ENCLAVE_FUNCTION
+**
+**==============================================================================
+*/
+
+#define OE_UNSUPPORTED_ENCLAVE_FUNCTION "unsupported enclave function"
+
+/*
+**==============================================================================
+**
+** <openssl/ssl.h>
+**
+**==============================================================================
+*/
+
+/*
+ * The following APIs allow users to configure an OpenSSL application to lookup
+ * certificates from the host filesystem, which is not trusted in OE's
+ * threat model, and therefore are recommended not to use inside an enclave.
+ */
+
+OE_OPENSSL_DEPRECATED(OE_UNSUPPORTED_ENCLAVE_FUNCTION)
```

Comment:
Why are these functions only deprecated and not removed? 
Do we expect some users to use these functions?